applications/new_schema.py

```diff
@@ -0,0 +1,75 @@
+import django_filters
+import graphene
+from graphene_django.filter import DjangoFilterConnectionField
+from graphene_django.types import DjangoObjectType
+from graphql_jwt.decorators import login_required, superuser_required
+
+from .models import BerthApplication, BerthSwitch, HarborChoice
+from .schema import BerthSwitchType as OldBerthSwitchType
+
+
+class HarborChoiceType(DjangoObjectType):
+    harbor = graphene.String()
+
+    class Meta:
+        model = HarborChoice
+        exclude = ("id", "application")
+
+    def resolve_harbor(self, info, **kwargs):
+        if self.harbor:
+            return self.harbor.servicemap_id
+        return None
+
+
+class BerthSwitchType(OldBerthSwitchType):
+    harbor = graphene.String()
+
+    class Meta:
+        model = BerthSwitch
+
+    def resolve_harbor(self, info, **kwargs):
+        if self.harbor:
+            return self.harbor.servicemap_id
+        return None
+
+
+class BerthApplicationFilter(django_filters.FilterSet):
+    has_berth_switch = django_filters.BooleanFilter(
+        field_name="berth_switch", method="filter_berth_switch"
+    )
+
+    def filter_berth_switch(self, queryset, name, value):
+        lookup = "__".join([name, "isnull"])
+        return queryset.filter(**{lookup: not value})
+
+
+class BerthApplicationTypeNode(DjangoObjectType):
+    boat_type = graphene.String()
+    harborchoice_set = graphene.List(HarborChoiceType)
+
+    class Meta:
+        model = BerthApplication
+        interfaces = (graphene.relay.Node,)
+        exclude = ("chosen_harbors",)
+
+    def resolve_boat_type(self, info, **kwargs):
+        if self.boat_type:
+            return self.boat_type.id
+        return None
+
+    def resolve_harborchoice_set(self, info, **kwargs):
+        if self.harborchoice_set.count():
+            return self.harborchoice_set.all()
+        return None
+
+
+class Query:
+    application = graphene.relay.Node.Field(BerthApplicationTypeNode)
+    applications = DjangoFilterConnectionField(
```

Comment:
also, keeping in mind that there will also be `winter_storage_applications`, could these be `berth_applications`? 🙂